### ISSUE TITLE
Node.js 13.14.0 'No "exports" main resolved in' @babel/helper-compilation-targets/package.json

### ISSUE THREAD

**Issue by JustFly1984** [formium/tsdx#900]
### Current Behavior
Node 13.3.0 - issues on CI/CD `browserlist@^4.14.5` dependency of dependency, breaking netlify build of our docs and examples by requiring node engine version newer than 13.7.0

tried both last versions of tsdx@0.13.3 v0.14.0, and experiencing different bugs in CI/CD.

v0.13.3
```
2:28:59 PM: ✓ Creating entry file 2.1 secs
2:28:59 PM: (babel plugin) Error: No "exports" main resolved in /opt/build/repo/packages/react-google-maps-api-marker-clusterer/node_modules/@babel/helper-compilation-targets/package.json
2:28:59 PM: Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main resolved in /opt/build/repo/packages/react-google-maps-api-marker-clusterer/node_modules/@babel/helper-compilation-targets/package.json
2:28:59 PM:     at applyExports (internal/modules/cjs/loader.js:487:9)
2:28:59 PM:     at resolveExports (internal/modules/cjs/loader.js:503:23)
2:28:59 PM:     at Function.Module._findPath (internal/modules/cjs/loader.js:631:31)
2:28:59 PM:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:949:27)
2:28:59 PM:     at Function.Module._load (internal/modules/cjs/loader.js:838:27)
2:28:59 PM:     at Module.require (internal/modules/cjs/loader.js:1022:19)
2:28:59 PM:     at require (internal/modules/cjs/helpers.js:72:18)
2:28:59 PM:     at Object.<anonymous> (/opt/build/repo/packages/react-google-maps-api-marker-clusterer/node_modules/@babel/preset-env/lib/debug.js:8:33)
2:28:59 PM:     at Module._compile (internal/modules/cjs/loader.js:1118:30)
2:28:59 PM:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1138:10)
```

v0.14.0
```
$ npx tsdx build --name infoBox --format=cjs,esm,umd
✓ Creating entry file 1.3 secs
(typescript) Error: ENOENT: no such file or directory, rename '/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache_' -> '/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache'
Error: ENOENT: no such file or directory, rename '/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache_' -> '/Users/justfly/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache'
    at Object.renameSync (fs.js:754:3)
    at RollingCache.roll (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24003:12)
    at TsCache.done (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24845:24)
    at Object._ongenerate (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:29098:21)
    at Object.generateBundle (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:29076:18)
    at /projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/rollup/dist/shared/node-entry.js:13117:25
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

error Command failed with exit code 1.
```

### Expected behavior

Should build packages with any node version.

### Suggested solution(s)

Fix build pipeline

### Additional context

You can fork https://github.com/JustFly1984/react-google-maps-api, run `nvm use` in each package directory separate, to test that each package builds correctly with combination tsdx@0.13.3 and node v13.3.0

But if you run `nvm use` in the root of repository (v13.14.0), and try to run build scripts from root package.json to build docs and gatsby.js example, you'll get errors^^^

### Your environment

```text
System:
    OS: macOS 10.15.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 238.93 MB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 13.14.0 - ~/.nvm/versions/node/v13.14.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v13.14.0/bin/npm
  Browsers:
    Chrome: 85.0.4183.121
    Chrome Canary: 88.0.4288.0
    Firefox: 79.0
    Safari: 14.0
  npmPackages:
    tsdx: 0.13.3 => 0.13.3 
    typescript: 4.0.3 => 4.0.3 
```
PS. My project is in condition where I can build and publish new versions of the lib, but can't deploy docs and examples.

**Comment by agilgur5** [comment]
@JustFly1984 you've duplicated your own #630 -- as you wrote there the issue is a bug in Node itself. Since then Node 13 reached EOL in June (c.f. #880).

Regarding the TSDX v0.14.0 issue, that duplicates #888 and is upstream in rpts2 as the error suggests . #892 duplicated it as well and pointed out it only seems to happen with UMD for some reason (which #898, #896 and this seem to confirm).
#896 is a PR to update to a recently patched version of rpts2 which I linked to in the duplicates, but as I wrote there, I'm waiting for another new version to be released that includes another bugfix that I personally made upstream.

**Comment by JustFly1984** [comment]
@agilgur5 Thank you for fast response!

I can't redeploy docs since June. My lib is used by 2500 projects, and growing. I want to start recording video tutorials. 
Can you please estimate how long it will take for next release? Would love to update node to latest 14 version. If you can confirm that tsdx@0.14.0 supports node@14, I will test node 14. Still not sure that gatsby.js supports node 14 though.

Have a great day!

**Comment by agilgur5** [comment]
> Can you please estimate how long it will take for next release?

ezolenko/rollup-plugin-typescript2#221 and #823  were the last two things I was looking to get in before releasing TSDX v0.14.1, but given ezolenko/rollup-plugin-typescript2#221 hasn't been released in over a week since merging and this UMD cache issue seems somewhat widespread, I was considering cutting a release for TSDX v0.14.1 early, possibly today.

But you should be able to use TSDX v0.13.3 fine. This bug is a bug in EOL Node 13, not in TSDX. 

> Still not sure that gatsby.js supports node 14 though

You can also downgrade to Node 12. Not sure if you're using some new Node feature or something.

> If you can confirm that tsdx@0.14.0 supports node@14

Yes, the internal PR I linked to regarding Node 13, #880, upgraded TSDX's internal CI matrix to use Node 14. So can confirm it supports Node 14. I think a number of users have confirmed that as well.

**Comment by JustFly1984** [comment]
@agilgur5  I have tested tsdx@0.13.3 and tsdx@0.14.0 with node 14.13.1 and it fails the build with same error

```
(typescript) Error: ENOENT: no such file or directory, rename /projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache_' -> '/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache'
Error: ENOENT: no such file or directory, rename /projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache_' -> '/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/.cache/rollup-plugin-typescript2/rpt2_2c34c68546ed43d39898c659568a2027b2495a19/code/cache'
    at Object.renameSync (fs.js:770:3)
    at RollingCache.roll (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24003:12)
    at TsCache.done (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:24845:24)
    at Object._ongenerate (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:29098:21)
    at Object.generateBundle (/projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:29076:18)
    at /projects/react-google-maps-api/packages/react-google-maps-api-infobox/node_modules/rollup/dist/shared/node-entry.js:13117:25
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

error Command failed with exit code 1.
```

**Comment by agilgur5** [comment]
> 0.13.3

Well actually this is a slightly different error than the one in Node 13, it says "defined" instead of "resolved".
In any case, I'd encourage you to read your own #630 as this is either a bug in Babel or in Node or both. It's quite upstream of TSDX and hit by 1000s of people, probably the most widespread Node bug I've seen. It's not in or specific to TSDX, so this isn't the right place

I looked into it for you and the upstream issue you linked to indeed suggests both are to blame and that `@babel/helper-compilation-targets`@7.8.3 has this bug. Your [`yarn.lock` pins that exact version](https://github.com/JustFly1984/react-google-maps-api/blob/df775fe3b086346a1d620c447f2d0e5e35c01593/packages/react-google-maps-api-infobox/yarn.lock#L77). You can upgrade `@babel/preset-env` in your lockfile to fix it.

> 0.14.0

Yes, I've already written and linked to where this error comes from and which issues it duplicates. There's quite a lot of details in those issues. That's not unique to Node version.

If you pass this error, you might still hit the error above. Though TSDX v0.14.0 upgraded the internal `@babel/preset-env` (#838 ), so it may indirectly change your `yarn.lock` and indirectly upgrade `@babel/helper-compilation-targets` to a newer version that doesn't have this bug.

**Comment by agilgur5** [comment]
The v0.14.0 cache issue has been fixed with #896 / https://github.com/ezolenko/rollup-plugin-typescript2/pull/243  and just released in [v0.14.1](https://github.com/formium/tsdx/releases/tag/v0.14.1).

You should be able to get v0.13.3 to work fine though, as I wrote above that seems to be directly due to your `yarn.lock` pin.
